File: Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int countLetters(string text)
{
    int countL = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            countL++;
        }
    }
    return countL;
}
int countWords(string text)
{
    int countW = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            countW++;
        }
    }
    return countW + 1;
}
int countSent(string text)
{
    int countS = 0;
    for (int j = 0; j < strlen(text) + 1; j++)
    {

        if (text[j] == '.' || text[j] == '!' || text[j] == '?')
        {
            countS++;
        }
    }
    return countS;
}
int clindex(float a, float b, float c)
{
    float L;
    float S;
    float Index;
    L = ((a / b) * 100.);
    S = ((c / b) * 100.);
    Index = 0.0588 * L - 0.296 * S - 15.8;
    return round(Index);
}
int main()
{
    string text = get_string("Text:");
    float a = countLetters(text); // countL
    float b = countWords(text);   // countW
    float c = countSent(text);    // countS
    int index = clindex(a, b, c);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

File: readabilityp/readability.py
from cs50 import get_string
import re

def count_letters(text):
    return len([c for c in text if c.isalpha()])

def count_words(text):
    return len(text.split())

def count_sentences(text):
    return len(re.findall(r'[.!?]', text))

def compute_cl_index(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100
    return round(0.0588 * L - 0.296 * S - 15.8)

text = get_string("Text: ")
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

index = compute_cl_index(letters, words, sentences)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
